feat: Add class for let's encrypt management

dadolhay opened the pull request:
Adds a class (to be extended later) that internally instantiates an acme client with the appropriate config.

Closes #17

humphd commented on CONTRIBUTING.md:
```diff
@@ -32,7 +32,7 @@ Optional:
 
 # Start MySQL in docker
-$ docker compose up -d
+$ docker-compose up -d
```
Docker bundles `compose` into `docker` now, so you can skip using the Python version.

dadolhay replied:
Interesting, under ubuntu 22.04 that command was unknown. I'll remove this change then

humphd replied:
Not all Linux distros install this for some reason, see https://docs.docker.com/compose/install/linux/

humphd commented on resources/LetsEncrypt.js:
```diff
@@ -0,0 +1,38 @@
+import acme from "acme-client";
```
Let's write this in TypeScript, and move it to `app/lib/lets-encrypt.server.ts`, similar to what @Genne23v is doing in #27.

Genne23v replied:
I just learned that we should name the file in a certain way in Remix. In case, you don't find this link in my long threaded code review. [Remix docs on server code and bundles](https://remix.run/docs/en/v1/pages/gotchas#server-code-in-client-bundles)

humphd reviewed (changes requested):
This file needs to be in `app/lib/lets-encrypt.server.ts` so it's reachable via the rest of the Remix backend.

humphd commented:
Also, the lock file needs an update.

dadolhay commented:
> This file needs to be in `app/lib/lets-encrypt.server.ts` so it's reachable via the rest of the Remix backend.

Ah, I renamed it, created the folder and forgot to move... my bad, sorry
Rebased to lastest origin/main, should have no conflicts now

humphd commented on app/lib/lets-encrypt.server.ts:
```diff
@@ -0,0 +1,39 @@
+import type { Client as AcmeClient } from "acme-client";
+import { secrets } from "docker-secret";
+
+const { LETSENCRYPT_ACCOUNT_PRIVATE_KEY_PEM } = secrets ?? {};
```
`LETS_ENCRYPT_...`

dadolhay replied:
Okay, updated

humphd reviewed (changes requested):
One last small thing, this is good after that.

Genne23v reviewed (approved):
It looks good to go
